GUIMainWindow.py

```diff
@@ -163,70 +173,90 @@ def drawLines(self):
         self.drawLines()
         self.canvas.update_idletasks
 
+    #removes a line from the canvas
     def deleteLine(self, firstClassName, secondClassName):
         del self.lineDict[(firstClassName, secondClassName)]
         self.drawLines()
 
+    #changes the type of a line on the canvas
     def renameLine(self, firstClassName, secondClassName, typ):
         del self.lineDict[(firstClassName, secondClassName)]
         self.drawLines()
         self.addLine(firstClassName, secondClassName, typ)
         self.drawLines()
 
+    #finds the shortest distance between two classes
     def findShortestDistance(self, firstClassName, secondClassName):
         # Attain list of Snaps from the classes
         firstClassSnaps = self.classDict[firstClassName].widgetCoordinates
         secondClassSnaps = self.classDict[secondClassName].widgetCoordinates
         self.shortestDistance = 1000000
         firstClassCoord = 0
         secondClassCoord = 0
-        secondIteration = 0
+        
+        destIteration = 0
 
         #iterate through all pairs of coordinates 
+        srcIteration = 0
         for coord1 in firstClassSnaps:
-            secondIteration = 0
+            destIteration = 0
             for coord2 in secondClassSnaps: 
                  
                 #calculate distance between points              
                 XDis = coord1[0] - coord2[0]
                 YDis = coord1[1] - coord2[1]
                 self.currentDistance = abs(YDis) + abs(XDis)
                 #check if any previous points were closer, it not. Replace it with the current distance
-                if (round(self.currentDistance) + 7 < self.shortestDistance):
-                    iterationSide = secondIteration
+                if (round(self.currentDistance) + 3 < self.shortestDistance):
+                    destSide = destIteration
+                    srcSide = srcIteration
                     self.shortestDistance = self.currentDistance
                     firstClassCoord = coord1
                     secondClassCoord = coord2
-                secondIteration = secondIteration + 1
 
-        print(iterationSide)
-        if(iterationSide == 1 or iterationSide == 4 or iterationSide == 6):
-            return [(firstClassCoord, secondClassCoord), 'left']
-        
-        if(iterationSide == 2 or iterationSide == 5 or iterationSide == 7):
-            return [(firstClassCoord, secondClassCoord), 'right']
-
-        if(iterationSide == 0):
-            
-            return [(firstClassCoord, secondClassCoord), 'top']
-        
+                destIteration = destIteration + 1
+            srcIteration = srcIteration + 1
         #return the closet points between the two classes
-        return [(firstClassCoord, secondClassCoord), 'bottom']
+        return [(firstClassCoord, secondClassCoord), self.sideToString(destSide), self.sideToString(srcSide) ]
+
+    def sideToString(self, sideNumber):
```

Comment:
In the future I'd probably make a map or list so you don't have to do any comparisons, like
```py
returnSide = ['top', 'left', 'right', 'bottom', 'left', 'right', 'left', 'right']
return returnSide[sideNumber]
```
or something similar to that